File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/WebViewClientSwazzledHooks.java
package io.embrace.android.embracesdk;

import androidx.annotation.Nullable;

import io.embrace.android.embracesdk.annotation.InternalApi;

/**
 * @hide
 */
@InternalApi
public final class WebViewClientSwazzledHooks {

    private WebViewClientSwazzledHooks() {
    }

    @SuppressWarnings("MethodNameCheck")
    public static void _preOnPageStarted(@Nullable android.webkit.WebView view,
                                         @Nullable java.lang.String url,
                                         @Nullable android.graphics.Bitmap favicon) {
        Embrace.getInstance().logWebView(url);
    }
}

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/WebViewClientSwazzledHooks.kt
package io.embrace.android.embracesdk

import android.graphics.Bitmap
import android.webkit.WebView
import io.embrace.android.embracesdk.annotation.InternalApi

/**
 * @hide
 */
@InternalApi
public object WebViewClientSwazzledHooks {

    @Suppress("UNUSED_PARAMETER")
    @JvmStatic
    public fun _preOnPageStarted(
        view: WebView?,
        url: String?,
        favicon: Bitmap?,
    ) {
        Embrace.getInstance().logWebView(url)
    }
}
